src/bindings/python/_actor.py

```diff
@@ -0,0 +1,60 @@
+import json
+from typing import Union, Callable
+
+try:
+    from .python_valhalla import _Actor
+except ModuleNotFoundError:
+    from python_valhalla import _Actor
+
+
+def _wrapper(func: Callable, req: Union[str, dict]) -> Union[str, dict]:
+    # return the type being passed, str -> str, dict -> dict
+    if isinstance(req, dict):
+        return json.loads(func(json.dumps(req)))
+    elif not isinstance(req, str):
+        raise ValueError(f"Request must be either of type str or dict")
+
+    return func(req)
+
+
+class Actor(_Actor):
+
+    def Route(self, req: Union[str, dict]) -> Union[str, dict]:
```

Comment:
subclass c++ `actor` so we have some more flexibility on the python side which c++ doesn't need to care about